server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -627,12 +627,12 @@ public String apply(Sink input)
 
         synchronized (handoffCondition) {
           while (!sinks.isEmpty()) {
-            handoffCondition.wait();
+            handoffCondition.wait(config.getHandoffConditionTimeout());
           }
         }
       }
       catch (InterruptedException e) {
-        throw Throwables.propagate(e);
+        throw new ISE("Failed to wait for segment handoff.", e);
```

Comment:
Actually, nm, this looks ok. Should be re-setting the interrupted state of the thread though, do you mind fixing that?